src/value/element_stream_reader.rs

```diff
@@ -204,19 +193,18 @@ impl IonReader for ElementStreamReader {
     }
 
     fn read_bool(&mut self) -> IonResult<bool> {
-        match self.current_value.as_ref() {
-            Some(element) if element.as_bool().is_some() => Ok(element.as_bool().unwrap()),
-            _ => Err(self.expected("bool value")),
-        }
+        self.current_value
+            .as_ref()
```

Comment:
It occurs to me that we could make a helper method like this:

```rust
fn current_value_as<T, F>(&self, expect_message: &'static str, map_fn: F) -> IonResult<T>
    where F: Fn(&Element) -> IonResult<T> {
    self.current_value.as_ref()
        .and_then(map_fn)
        .ok_or_else(|| self.expected(expect_message))
}
// ^-- Sorry for any errors, it's hard to write code in GitHub comments 😬 
```

Which would let us simplify all of these to (e.g.):
```rust
self.current_value_as("bool_value", |v| v.as_bool())
```